src/library/scala/math/Ordering.scala

```diff
@@ -214,9 +232,18 @@ object Ordering extends LowPriorityOrderingImplicits {
 
   @inline def apply[T](implicit ord: Ordering[T]) = ord
 
+  /** An ordering which caches the value of its reverse. */
+  trait CachedReverse[T] extends Ordering[T] {
```

Comment:
probably make this `sealed`

```suggestion
  sealed trait CachedReverse[T] extends Ordering[T] {
```

Reply:
Why not make it private as well?

Reply:
I tried that and it doesn't work. It escapes because `Ordering.Int` is public and extends it.